docs/validation.md

```diff
@@ -0,0 +1,93 @@
+# NIX file validity
+
+## Hard vs Soft rules
+
+The NIXIO libraries enforce the general NIX structure and requirements.  For instance, all data objects (Block, Group, DataArray, etc) must always have a name and a type and all the libraries and tools that can create NIX files enforce these rules.  Beyond the rules that the library enforces, there are also certain that are the responsibility of the user to follow.  These rules are not required in a working NIX file, but violating them is often indication that a file is _semantically invalid_ or simply incomplete.  We can separate all the validity rules into two categories, _hard_ and _soft_ rules, depending on their severity.
+
+_Hard rules_ include the ones enforced by the library mentioned above, but also some rules that the users are expected to follow.  An example of these user-dependent _hard rules_ is that the number of Dimension objects attached to a DataArray must match the number of dimensions in the data itself.  Violating this rule makes the DataArray _semantically invalid_, but it does not prevent the use of the file in any way.
+
+_Soft rules_ are less important than _hard rules_, as the name suggests.  Violating them does not make a file invalid, neither functionally nor semantically, but may limit certain functionality in the library or may be an indication of incomplete data descriptions.  An example of a _soft rule_ is the absence of `units` in a DataArray, or the use of a non-SI unit.  This rule violation prevents the acquisition of data from a DataArray through a Tag or MultiTag object when the units of the objects are scaled versions of the each other.
+
+This document lists both the _hard rules_ and _soft rules_ that are required for creating semantically valid and complete NIX files, separated by object type.
+
+## The NIX validator
+
+The NIXIO libraries implement a validator that can check a file for both soft and hard rule violations and inconsistencies.  The output of these validators label hard rule violations as _errors_ and soft rule violations as _warnings_.
+
+## Hard rules
+
+### File objects
+- Creation date must be set: HDF5 attribute `created_at`.
+
```

Comment:
Don't we enforce format, and version attributes?

Reply:
Right.  I think I skipped them because the file wont even open without them, but they should be mentioned.

Reply:
Oh no, I just realised.  Format and Version appear in the Soft Rules below.

Here they're marked as `should`:
https://github.com/G-Node/nix/blob/789e1bc802fd22470c0c5e54c789400408ee5d4b/src/valid/validate.cpp#L233-L234